src/main/java/com/microsoft/spring/data/gremlin/conversion/result/GremlinResultEdgeReader.java

```diff
@@ -21,6 +21,17 @@ public GremlinResultEdgeReader() {
         super();
     }
 
+    private void readProperties(@NonNull Map<String, Object> properties, @NonNull GremlinSource source) {
+        Assert.isTrue(source.getProperties().isEmpty(), "should be empty GremlinSource");
+
+        for (final Map.Entry<String, Object> entry : properties.entrySet()) {
```

Comment:
I suppose we used Java 8. Then I would prefer using lambda for this. Something like:
```java
properties.forEach((k ,v) -> source.setProperty(k, v))
```

Reply:
copy that.